src/test/java/uk/ac/ebi/ampt2d/metadata/MetadataApplicationTest.java

```diff
@@ -227,32 +255,43 @@ private String postTestStudy(String accession, int version, String name, String
 
     @Test
     public void postAnalysis() throws Exception {
-        String referenceSequenceUrl = postTestReferenceSequence("GRCh37", "p2",
+        String assemblyUrl = postTestAssembly("GRCh37", "p2",
                 Arrays.asList("GCA_000001405.3", "GCF_000001405.14"));
+        String transcriptomeUrl = postTestTranscriptome("GRCh38", "p3",
+                Arrays.asList("GCA_000001405.4", "GCF_000001405.15"));
+        List<String> geneUrlList = new ArrayList<String>();
+        String geneUrl = postTestGene("GRCh39", "p4",
+                Arrays.asList("GCA_000001405.5", "GCF_000001405.16"));
+        geneUrlList.add(geneUrl);
+        geneUrl = postTestGene("GRCh40", "p5",
+                Arrays.asList("GCA_000001405.5", "GCF_000001405.16"));
+        geneUrlList.add(geneUrl);
         String studyUrl = postTestStudy("EGAS0001", 1, "test_human_study");
 
-        String location = postTestAnalysis("EGAA0001", referenceSequenceUrl, studyUrl, Analysis.Technology.GWAS,
+        String location = postTestAnalysis("EGAA0001", assemblyUrl, transcriptomeUrl, geneUrlList, studyUrl, Analysis.Technology.GWAS,
```

Comment:
Can you also test the cases in which an analysis is based on only one type of reference sequence? In fact, I think for now we should disallow an analysis based on three different types of reference sequence. @cyenyxe

Reply:
I will verify if T2D-218 changes can help here.